docs/integration-guides/ipc-integration.md

```diff
@@ -0,0 +1,288 @@
+
+### Call example
+
+```bash
+curl --header "Nano-Api-Key:mywalletuser" --insecure -d \
+   '{ "account": "nano_3t6k35gi95xu6tergt6p69ck76ogmitsa8mnijtpxm9fkcm736xtoncuohr3"}' \
+   https://www.example.com:7076/api/v2/AccountWeight
+```
+This uses HTTPS (which the node supports through a build option), and the `--insecure` is there because the node's certificate in this example is self-signed.
+
+Using an envelope instead of the `AccountWeight` endpoint:
+
+```json
+{ 
+   "credentials": "mywalletuser",
+   "message_type" : "AccountWeight", 
+   "message": 
+   {
+       "account": "nano_3t6k35gi95xu6tergt6p69ck76ogmitsa8mnijtpxm9fkcm736xtoncuohr3" 
+   }
+} 
+```
+
+`POST` the above to https://www.example.com:7076/api/v2
+
+### Configuration examples
+
+For testing IPC without caring about permissions, this gives access to everything:
+
+```toml
+[[user]]
+allow = "unrestricted"
+```
+
+A more elaborate sample:
+
+```toml
+[[role]]
+id = "process_admin"
+allow = "api_service_stop"
+
+[[role]]
+id = "walletadmin"
+allow = "wallet_read, wallet_write"
+
+# Add some additional permissions to the default user
+# This is used when no credentials are given
+[[user]]
+allow = "wallet_seed_change, epoch_upgrade, wallet_read"
+
+# The default user can also be set to bare (no default permissions). 
+# That way, a node can be exposed with a limited set of default permissions.
+#bare = true
+
+[[user]]
+id = "mywalletuser"
+# Inherit all the permissions from these roles
+roles = "process_admin, walletadmin"
+
+# Add additional permissions for this specific user
+allow = "wallet_seed_change, epoch_upgrade"
+
+# Default and inherited permissions can be denied
+deny = "wallet_read"
+
+[[user]]
+id = "epoch-user"
+
+# Add specific permission for this user
+allow = "epoch_upgrade"
+
+# Do not inherit any default permissions. This is useful
+# for making users with an explicit set of minimum permissions.
+bare = true
+```
+
+### Reload config
+
+The access file can be reloaded without restarting the node or wallet. For the node:
+
+`killall -SIGHUP nano_node`
+
+(actual syntax depends on OS)
+
+---
+
+## IPC V1 Details
+
+??? info
+
+	As of v18, the Nano node exposes a low level IPC interface over which multiple future APIs can be marshalled. Currently, the IPC interface supports the legacy RPC JSON format. The HTTP based RPC server is still available. Because the only IPC encoding is currently "legacy RPC", RPC config options like "enable_control" still apply.
+
+	**Transports**
+
+	TCP and unix domain sockets are supported. Named pipes and shared memory may be supported in future releases.
+
+	**IPC clients**
+
+	A demo web server written in Go is available at https://github.com/nanocurrency/rpc-go. This allows HTTP clients to make JSON requests via IPC, which is compatible with the existing format. The web server can communicate with a node over domain sockets or TCP.
+
```

Comment:
I think we can remove this section now, it's not used and IPC 2.0 already supports this for RPC 2.0

Reply:
Removed